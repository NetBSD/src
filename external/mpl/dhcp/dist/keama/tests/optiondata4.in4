# option data config

# options
option space foobar;

option ip-forwarding on;

option foobar.fmt-b-si8 code 3 = signed integer 8;
option foobar.fmt-b-si8 -100;

option default-ip-ttl 20;

option foobar.fmt-s-si16 code 6 = signed integer 16;
option foobar.fmt-s-si16 -1000;

option boot-size 16000;

option time-offset -1200;

option path-mtu-aging-timeout 86400;

option swap-server 10.5.5.1;

option foobar.fmt-6 code 12 = ip6-address;
option foobar.fmt-6 2001::1;

option foobar.fmt-d code 13 = domain-name;
# Silly, d aka domain-name are without quotes, D aka domain-list are with
option foobar.fmt-d www.example.com;

option bcms-controller-names "foo.bar", "www.no-where.biz";

option domain-search "example.com", "example.org";

option tftp-server-name "my-server";

option dhcp-client-identifier 01:02:aa:bb;

option foobar.fmt-Z code 18 = zerolen;
option foobar.fmt-Z;

option foobar.fmt-Ba code 50 = array of unsigned integer 8;
option dhcp-parameter-request-list 1, 2, 3;

option foobar.fmt-fB code 100 = { boolean, unsigned integer 8 };
option foobar.fmt-fB off 66;

option routers 10.5.5.1, 10.5.5.2, 10.5.5.3;
