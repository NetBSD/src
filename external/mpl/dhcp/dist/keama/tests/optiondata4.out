{
  # option data config
  # options
  "Dhcp4": {
    "option-data": [
      {
        "space": "dhcp4",
        "name": "ip-forwarding",
        "code": 19,
//      "original-data": "on",
        /// canonized booleans to lowercase true or false
        "data": "true"
      },
      {
        "space": "foobar",
        "name": "fmt-b-si8",
        "code": 3,
        "data": "-100"
      },
      {
        "space": "dhcp4",
        "name": "default-ip-ttl",
        "code": 23,
        "data": "20"
      },
      {
        "space": "foobar",
        "name": "fmt-s-si16",
        "code": 6,
        "data": "-1000"
      },
      {
        "space": "dhcp4",
        "name": "boot-size",
        "code": 13,
        "data": "16000"
      },
      {
        "space": "dhcp4",
        "name": "time-offset",
        "code": 2,
        "data": "-1200"
      },
      {
        "space": "dhcp4",
        "name": "path-mtu-aging-timeout",
        "code": 24,
        "data": "86400"
      },
      {
        "space": "dhcp4",
        "name": "swap-server",
        "code": 16,
        "data": "10.5.5.1"
      },
      {
        "space": "foobar",
        "name": "fmt-6",
        "code": 12,
        "data": "2001::1"
      },
      # Silly, d aka domain-name are without quotes, D aka domain-list are with
      {
        "space": "foobar",
        "name": "fmt-d",
        "code": 13,
        "data": "www.example.com"
      },
      {
        "space": "dhcp4",
        "name": "bcms-controller-names",
        "code": 88,
//      "original-data": "\"foo.bar\", \"www.no-where.biz\"",
        "data": "foo.bar, www.no-where.biz"
      },
      {
        "space": "dhcp4",
        "name": "domain-search",
        "code": 119,
//      "original-data": "\"example.com\", \"example.org\"",
        "data": "example.com, example.org"
      },
      {
        "space": "dhcp4",
        "name": "tftp-server-name",
        "code": 66,
        "data": "my-server"
      },
      {
        "space": "dhcp4",
        "name": "dhcp-client-identifier",
        "code": 61,
//      "original-data": "01:02:aa:bb",
        "csv-format": false,
        "data": "0102aabb"
      },
      {
        "space": "foobar",
        "name": "fmt-Z",
        "code": 18,
        "data": ""
      },
      /// Possible PRL hack
      /// Consider setting "always-send" to true when setting data for relevant options, cf Kea #250
      {
        "space": "dhcp4",
        "name": "dhcp-parameter-request-list",
        "code": 55,
        "data": "1, 2, 3"
      },
      {
        "space": "foobar",
        "name": "fmt-fB",
        "code": 100,
//      "original-data": "off 66",
        /// canonized booleans to lowercase true or false
        "data": "false, 66"
      },
      {
        "space": "dhcp4",
        "name": "routers",
        "code": 3,
        "data": "10.5.5.1, 10.5.5.2, 10.5.5.3"
      }
    ],
    "option-def": [
      {
        "space": "foobar",
        "name": "fmt-b-si8",
        "code": 3,
        "type": "int8"
      },
      {
        "space": "foobar",
        "name": "fmt-s-si16",
        "code": 6,
        "type": "int16"
      },
      {
        "space": "foobar",
        "name": "fmt-6",
        "code": 12,
        "type": "ipv6-address"
      },
      {
        "space": "foobar",
        "name": "fmt-d",
        "code": 13,
        "type": "fqdn"
      },
      {
        "space": "foobar",
        "name": "fmt-Z",
        "code": 18,
        "type": "empty"
      },
      {
        "space": "foobar",
        "name": "fmt-Ba",
        "code": 50,
        "array": true,
        "type": "uint8"
      },
      {
        "space": "foobar",
        "name": "fmt-fB",
        "code": 100,
        "record-types": "boolean, uint8",
        "type": "record"
      }
    ]
  }
}
